File: backend/laissez/src/main/java/com/pdsi/laissez/models/Produto.java
package com.pdsi.laissez.models;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import com.fasterxml.jackson.annotation.JsonBackReference;

@Entity
public class Produto {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long idproduto;
	
	@JsonBackReference
	@ManyToOne
	@JoinColumn(name="supermercado_idsupermercado")
	private Supermercado supermercado;
	
	@Column
	private String nome;
	
	@Column
	private String marca;
	
	@Column
	private Long qrcode;
	
	@Column
	private String descricao;
	
	@Column
	private String tipo;
	
	@Column
	private Double preco;
	
	@Column
	private String imagem;

	public Long getIdproduto() {
		return idproduto;
	}

	public void setIdproduto(Long idproduto) {
		this.idproduto = idproduto;
	}

	public Supermercado getSupermercado() {
		return supermercado;
	}

	public void setSupermercado(Supermercado supermercado) {
		this.supermercado = supermercado;
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getMarca() {
		return marca;
	}

	public void setMarca(String marca) {
		this.marca = marca;
	}

	public Long getQrcode() {
		return qrcode;
	}

	public void setQrcode(Long qrcode) {
		this.qrcode = qrcode;
	}

	public String getDescricao() {
		return descricao;
	}

	public void setDescricao(String descricao) {
		this.descricao = descricao;
	}

	public String getTipo() {
		return tipo;
	}

	public void setTipo(String tipo) {
		this.tipo = tipo;
	}

	public String getImagem() {
		return imagem;
	}

	public void setImagem(String imagem) {
		this.imagem = imagem;
	}

	public Double getPreco() {
		return preco;
	}

	public void setPreco(Double preco) {
		this.preco = preco;
	}
	
	
}

File: frontend/laissez/lib/models/produto.dart
class Produto {
  final int idproduto;
  final String nome;
  final String marca;
  final String descricao;
  final String tipo;
  final double preco;
  final String imagem;
  final int qrcode;
  final int quantidade;

  Produto({
    this.idproduto,
    this.nome,
    this.marca,
    this.descricao,
    this.tipo,
    this.preco,
    this.imagem,
    this.qrcode,
    this.quantidade,
  });

  factory Produto.fromJson(Map<String, dynamic> json) {
    return Produto(
      idproduto: json['idproduto'],
      nome: json['nome'],
      marca: json['marca'],
      descricao: json['descricao'],
      tipo: json['tipo'],
      preco: json['preco'],
      imagem: json['imagem'],
      qrcode: json['qrcode'],
      quantidade: json['quantidade'],
    );
  }
}
